theano/gpuarray/opt.py

```diff
@@ -315,7 +328,12 @@ def apply(self, fgraph):
 
             # Move only if any of the inputs are on the GPU.
             move_to_GPU = False
+            if any([isinstance(i, GpuArrayVariable) or
+                   isinstance(i, GpuArraySharedVariable)
+                   for i in [mapping[v] for v in node.inputs] +
+                   node.outputs]):
 
+                move_to_GPU = True
```

Comment:
Exactly! I don't know how it came back. Probably i ignored during rebase. And yeah, that output part, @lamblin wanted to keep it as it might be helpful later when we implement backward pass